BENCH_DATA M4_COMP_DEMOGRAPHICS <Bench.TS_datasets.cTimeSeriesDatasetSpec object at 0x7f3e624e9860>
TIME :  Date N= 12 H= 6 HEAD= [0 1 2 3 4] TAIL= [ 7  8  9 10 11]
SIGNAL :  D0499 N= 12 H= 6 HEAD= [ 4.9  4.9  4.9  4.9  4.9] TAIL= [ 4.7  4.7  4.6  4.6  4.6]
   D0499  Date
0    4.9     0
1    4.9     1
2    4.9     2
3    4.9     3
4    4.9     4
<class 'pandas.core.frame.DataFrame'>
Int64Index: 12 entries, 0 to 11
Data columns (total 2 columns):
D0499    12 non-null float64
Date     12 non-null int64
dtypes: float64(1), int64(1)
memory usage: 288.0 bytes
None
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 140, in trainModel
    lAutoF1.train(df , self.mTSSpec.mTimeVar , iSignal, iHorizon)
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 25, in train
    self.mSignalDecomposition.train(iInputDS, iTime, iSignal, iHorizon, iExogenousData);
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 459, in train
    logger.info("START_TRAINING '" + iSignal + "'")
Message: "START_TRAINING 'D0499'"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 140, in trainModel
    lAutoF1.train(df , self.mTSSpec.mTimeVar , iSignal, iHorizon)
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 25, in train
    self.mSignalDecomposition.train(iInputDS, iTime, iSignal, iHorizon, iExogenousData);
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 481, in train
    logger.info("END_TRAINING_TIME_IN_SECONDS '" + iSignal + "' " + str(self.mTrainingTime))
Message: "END_TRAINING_TIME_IN_SECONDS 'D0499' 4.032837867736816"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 85, in getInfo
    logger.info("TIME_DETAIL " + self.mTrend.mTimeInfo.info());
Message: "TIME_DETAIL TimeVariable='Date' TimeMin=0 TimeMax=11 TimeDelta=1.0 Estimation = (0 , 12) Validation = (0 , 12) Test = (0 , 12) Horizon=6"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 86, in getInfo
    logger.info("SIGNAL_DETAIL " + self.info());
Message: "SIGNAL_DETAIL SignalVariable='_D0499' Min=4.6 Max=4.9  Mean=4.775 StdDev=0.123322071558"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 87, in getInfo
    logger.info("BEST_TRANSOFORMATION_TYPE '" + self.mTransformation.get_name("") + "'");
Message: "BEST_TRANSOFORMATION_TYPE '_'"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 88, in getInfo
    logger.info("BEST_DECOMPOSITION  '" + self.mOutName + "' [" + self.getFormula() + "]");
Message: "BEST_DECOMPOSITION  '_D0499_LinearTrend_residue_zeroCycle_residue_NoAR' [LinearTrend + NoCycle + NoAR]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 89, in getInfo
    logger.info("TREND_DETAIL '" + self.mTrend.mOutName + "' [" + self.mTrend.mFormula + "]");
Message: "TREND_DETAIL '_D0499_LinearTrend' [LinearTrend]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 90, in getInfo
    logger.info("CYCLE_DETAIL '"+ self.mCycle.mOutName + "' [" + self.mCycle.mFormula + "]");
Message: "CYCLE_DETAIL '_D0499_LinearTrend_residue_zeroCycle' [NoCycle]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 91, in getInfo
    logger.info("AUTOREG_DETAIL '" + self.mAR.mOutName + "' [" + self.mAR.mFormula + "]");
Message: "AUTOREG_DETAIL '_D0499_LinearTrend_residue_zeroCycle_residue_NoAR' [NoAR]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 92, in getInfo
    logger.info("MODEL_MAPE MAPE_Fit=" + str(self.mFitPerf.mMAPE) + " MAPE_Forecast=" + str(self.mForecastPerf.mMAPE)  + " MAPE_Test=" + str(self.mTestPerf.mMAPE) );
Message: 'MODEL_MAPE MAPE_Fit=0.0124 MAPE_Forecast=0.0124 MAPE_Test=0.0124'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 93, in getInfo
    logger.info("MODEL_MASE MASE_Fit=" + str(self.mFitPerf.mMASE) + " MASE_Forecast=" + str(self.mForecastPerf.mMASE)  + " MASE_Test=" + str(self.mTestPerf.mMASE) );
Message: 'MODEL_MASE MASE_Fit=2.1603 MASE_Forecast=2.1603 MASE_Test=2.1603'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 94, in getInfo
    logger.info("MODEL_L1 L1_Fit=" + str(self.mFitPerf.mL2) + " L1_Forecast=" + str(self.mForecastPerf.mL2)  + " L1_Test=" + str(self.mTestPerf.mL1) );
Message: 'MODEL_L1 L1_Fit=0.0661827945926 L1_Forecast=0.0661827945926 L1_Test=0.0589172979798'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 95, in getInfo
    logger.info("MODEL_L2 L2_Fit=" + str(self.mFitPerf.mL2) + " L2_Forecast=" + str(self.mForecastPerf.mL2)  + " L2_Test=" + str(self.mTestPerf.mL2) );
Message: 'MODEL_L2 L2_Fit=0.0661827945926 L2_Forecast=0.0661827945926 L2_Test=0.0661827945926'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 96, in getInfo
    logger.info("MODEL_COMPLEXITY " + str(self.getComplexity()) );
Message: 'MODEL_COMPLEXITY 16'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 97, in getInfo
    logger.info("AR_MODEL_DETAIL_START");
Message: 'AR_MODEL_DETAIL_START'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 330, in run_multiprocessed
    pool = mp.Pool(nbprocesses)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 57, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 153, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 141, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 132, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 99, in getInfo
    logger.info("AR_MODEL_DETAIL_END");
Message: 'AR_MODEL_DETAIL_END'
Arguments: ()
   Transformation                                              Model  \
0          _D0499    _D0499_Lag1Trend_residue_zeroCycle_residue_NoAR   
1          _D0499  _D0499_Lag1Trend_residue_bestCycle_byL2_residu...   
2      Diff_D0499  Diff_D0499_Lag1Trend_residue_zeroCycle_residue...   
3   RelDiff_D0499  RelDiff_D0499_Lag1Trend_residue_zeroCycle_resi...   
4      Diff_D0499  Diff_D0499_Lag1Trend_residue_bestCycle_byL2_re...   
5   RelDiff_D0499  RelDiff_D0499_Lag1Trend_residue_bestCycle_byL2...   
6          _D0499    _D0499_PolyTrend_residue_zeroCycle_residue_NoAR   
7          _D0499  _D0499_PolyTrend_residue_bestCycle_byL2_residu...   
8      Diff_D0499  Diff_D0499_LinearTrend_residue_zeroCycle_resid...   
9      Diff_D0499  Diff_D0499_LinearTrend_residue_bestCycle_byL2_...   
10     Diff_D0499  Diff_D0499_PolyTrend_residue_zeroCycle_residue...   
11     Diff_D0499  Diff_D0499_PolyTrend_residue_bestCycle_byL2_re...   
12     Diff_D0499  Diff_D0499_ConstantTrend_residue_zeroCycle_res...   
13     Diff_D0499  Diff_D0499_ConstantTrend_residue_bestCycle_byL...   
14  RelDiff_D0499  RelDiff_D0499_PolyTrend_residue_zeroCycle_resi...   
15  RelDiff_D0499  RelDiff_D0499_PolyTrend_residue_bestCycle_byL2...   
16  RelDiff_D0499  RelDiff_D0499_LinearTrend_residue_zeroCycle_re...   
17  RelDiff_D0499  RelDiff_D0499_LinearTrend_residue_bestCycle_by...   
18         _D0499  _D0499_LinearTrend_residue_zeroCycle_residue_NoAR   
19         _D0499  _D0499_LinearTrend_residue_bestCycle_byL2_resi...   
20  RelDiff_D0499  RelDiff_D0499_ConstantTrend_residue_zeroCycle_...   
21  RelDiff_D0499  RelDiff_D0499_ConstantTrend_residue_bestCycle_...   
22         _D0499  _D0499_ConstantTrend_residue_zeroCycle_residue...   
23         _D0499  _D0499_ConstantTrend_residue_bestCycle_byL2_re...   
24   CumSum_D0499  CumSum_D0499_Lag1Trend_residue_zeroCycle_resid...   
25   CumSum_D0499  CumSum_D0499_Lag1Trend_residue_bestCycle_byL2_...   
26   CumSum_D0499  CumSum_D0499_PolyTrend_residue_zeroCycle_resid...   
27   CumSum_D0499  CumSum_D0499_PolyTrend_residue_bestCycle_byL2_...   
28   CumSum_D0499  CumSum_D0499_LinearTrend_residue_zeroCycle_res...   
29   CumSum_D0499  CumSum_D0499_LinearTrend_residue_bestCycle_byL...   
30   CumSum_D0499  CumSum_D0499_ConstantTrend_residue_zeroCycle_r...   
31   CumSum_D0499  CumSum_D0499_ConstantTrend_residue_bestCycle_b...   

    Complexity  FitCount     FitL1     FitL2  FitMAPE   FitMASE  \
0           32        12  0.025000  0.050000   0.0053    0.9167   
1           40        12  0.025000  0.050000   0.0053    0.9167   
2           64        12  0.025000  0.050000   0.0053    0.9167   
3           64        12  0.025000  0.050000   0.0053    0.9167   
4           72        12  0.025000  0.050000   0.0053    0.9167   
5           72        12  0.025000  0.050000   0.0053    0.9167   
6           16        12  0.036164  0.043428   0.0076    1.3260   
7           24        12  0.036164  0.043428   0.0076    1.3260   
8           48        12  0.038636  0.047625   0.0080    1.4167   
9           56        12  0.038636  0.047625   0.0080    1.4167   
10          48        12  0.038825  0.047654   0.0081    1.4236   
11          56        12  0.038825  0.047654   0.0081    1.4236   
12          32        12  0.050000  0.062082   0.0103    1.8333   
13          40        12  0.050000  0.062082   0.0103    1.8333   
14          48        12  0.051393  0.059516   0.0107    1.8844   
15          56        12  0.051393  0.059516   0.0107    1.8844   
16          48        12  0.054157  0.062984   0.0113    1.9857   
17          56        12  0.054157  0.062984   0.0113    1.9857   
18          16        12  0.058917  0.066183   0.0124    2.1603   
19          24        12  0.058917  0.066183   0.0124    2.1603   
20          32        12  0.079760  0.091631   0.0165    2.9245   
21          40        12  0.079760  0.091631   0.0165    2.9245   
22           0        12  0.112500  0.123322   0.0237    4.1250   
23           8        12  0.112500  0.123322   0.0237    4.1250   
24          64        12  0.433333  1.415392   0.0887   15.8889   
25          72        12  0.433333  1.415392   0.0887   15.8889   
26          48        12  2.521374  3.733879   0.5201   92.4504   
27          56        12  2.521374  3.733879   0.5201   92.4504   
28          48        12  3.027691  4.130591   0.6292  111.0153   
29          56        12  3.027691  4.130591   0.6292  111.0153   
30          32        12  6.578472  8.917399   1.3681  241.2106   
31          40        12  6.578472  8.917399   1.3681  241.2106   

    ForecastCount  ForecastL1  ForecastL2  ForecastMAPE  ForecastMASE  \
0              12    0.025000    0.050000        0.0053        0.9167   
1              12    0.025000    0.050000        0.0053        0.9167   
2              12    0.025000    0.050000        0.0053        0.9167   
3              12    0.025000    0.050000        0.0053        0.9167   
4              12    0.025000    0.050000        0.0053        0.9167   
5              12    0.025000    0.050000        0.0053        0.9167   
6              12    0.036164    0.043428        0.0076        1.3260   
7              12    0.036164    0.043428        0.0076        1.3260   
8              12    0.038636    0.047625        0.0080        1.4167   
9              12    0.038636    0.047625        0.0080        1.4167   
10             12    0.038825    0.047654        0.0081        1.4236   
11             12    0.038825    0.047654        0.0081        1.4236   
12             12    0.050000    0.062082        0.0103        1.8333   
13             12    0.050000    0.062082        0.0103        1.8333   
14             12    0.051393    0.059516        0.0107        1.8844   
15             12    0.051393    0.059516        0.0107        1.8844   
16             12    0.054157    0.062984        0.0113        1.9857   
17             12    0.054157    0.062984        0.0113        1.9857   
18             12    0.058917    0.066183        0.0124        2.1603   
19             12    0.058917    0.066183        0.0124        2.1603   
20             12    0.079760    0.091631        0.0165        2.9245   
21             12    0.079760    0.091631        0.0165        2.9245   
22             12    0.112500    0.123322        0.0237        4.1250   
23             12    0.112500    0.123322        0.0237        4.1250   
24             12    0.433333    1.415392        0.0887       15.8889   
25             12    0.433333    1.415392        0.0887       15.8889   
26             12    2.521374    3.733879        0.5201       92.4504   
27             12    2.521374    3.733879        0.5201       92.4504   
28             12    3.027691    4.130591        0.6292      111.0153   
29             12    3.027691    4.130591        0.6292      111.0153   
30             12    6.578472    8.917399        1.3681      241.2106   
31             12    6.578472    8.917399        1.3681      241.2106   

    TestCount    TestL1    TestL2  TestMAPE  TestMASE  
0          12  0.025000  0.050000    0.0053    0.9167  
1          12  0.025000  0.050000    0.0053    0.9167  
2          12  0.025000  0.050000    0.0053    0.9167  
3          12  0.025000  0.050000    0.0053    0.9167  
4          12  0.025000  0.050000    0.0053    0.9167  
5          12  0.025000  0.050000    0.0053    0.9167  
6          12  0.036164  0.043428    0.0076    1.3260  
7          12  0.036164  0.043428    0.0076    1.3260  
8          12  0.038636  0.047625    0.0080    1.4167  
9          12  0.038636  0.047625    0.0080    1.4167  
10         12  0.038825  0.047654    0.0081    1.4236  
11         12  0.038825  0.047654    0.0081    1.4236  
12         12  0.050000  0.062082    0.0103    1.8333  
13         12  0.050000  0.062082    0.0103    1.8333  
14         12  0.051393  0.059516    0.0107    1.8844  
15         12  0.051393  0.059516    0.0107    1.8844  
16         12  0.054157  0.062984    0.0113    1.9857  
17         12  0.054157  0.062984    0.0113    1.9857  
18         12  0.058917  0.066183    0.0124    2.1603  
19         12  0.058917  0.066183    0.0124    2.1603  
20         12  0.079760  0.091631    0.0165    2.9245  
21         12  0.079760  0.091631    0.0165    2.9245  
22         12  0.112500  0.123322    0.0237    4.1250  
23         12  0.112500  0.123322    0.0237    4.1250  
24         12  0.433333  1.415392    0.0887   15.8889  
25         12  0.433333  1.415392    0.0887   15.8889  
26         12  2.521374  3.733879    0.5201   92.4504  
27         12  2.521374  3.733879    0.5201   92.4504  
28         12  3.027691  4.130591    0.6292  111.0153  
29         12  3.027691  4.130591    0.6292  111.0153  
30         12  6.578472  8.917399    1.3681  241.2106  
31         12  6.578472  8.917399    1.3681  241.2106  
  Transformation                                              Model  \
0         _D0499    _D0499_Lag1Trend_residue_zeroCycle_residue_NoAR   
1         _D0499  _D0499_Lag1Trend_residue_bestCycle_byL2_residu...   
2     Diff_D0499  Diff_D0499_Lag1Trend_residue_zeroCycle_residue...   
3  RelDiff_D0499  RelDiff_D0499_Lag1Trend_residue_zeroCycle_resi...   
4     Diff_D0499  Diff_D0499_Lag1Trend_residue_bestCycle_byL2_re...   

   Complexity  FitCount  FitL1  FitL2  FitMAPE  FitMASE  ForecastCount  \
0          32        12  0.025   0.05   0.0053   0.9167             12   
1          40        12  0.025   0.05   0.0053   0.9167             12   
2          64        12  0.025   0.05   0.0053   0.9167             12   
3          64        12  0.025   0.05   0.0053   0.9167             12   
4          72        12  0.025   0.05   0.0053   0.9167             12   

   ForecastL1  ForecastL2  ForecastMAPE  ForecastMASE  TestCount  TestL1  \
0       0.025        0.05        0.0053        0.9167         12   0.025   
1       0.025        0.05        0.0053        0.9167         12   0.025   
2       0.025        0.05        0.0053        0.9167         12   0.025   
3       0.025        0.05        0.0053        0.9167         12   0.025   
4       0.025        0.05        0.0053        0.9167         12   0.025   

   TestL2  TestMAPE  TestMASE  
0    0.05    0.0053    0.9167  
1    0.05    0.0053    0.9167  
2    0.05    0.0053    0.9167  
3    0.05    0.0053    0.9167  
4    0.05    0.0053    0.9167  
6 0    4.5
1    4.5
2    4.4
3    4.4
4    4.4
5    4.3
Name: D0499, dtype: float64
6 0    4.655587
1    4.637216
2    4.618845
3    4.600473
4    4.582102
5    4.563731
Name: D0499_Forecast, dtype: float64
FORECAST_DETAIL_ACTUAL D0499 D0499 [ 4.5  4.5  4.4  4.4  4.4  4.3]
FORECAST_DETAIL_PREDICTED D0499 D0499 [ 4.65558712  4.63721591  4.6188447   4.60047348  4.58210227  4.56373106]
BENCHMARK_PERF_DETAIL D0499 D0499 12 6 4.294032573699951 LinearTrend + NoCycle + NoAR 6 0.0438 0.0429 4.8248 0.192992424242 0.19741390004 -7.25294659115
BENCHMARK_SUCCESS 'M4_COMP_DEMOGRAPHICS_D0499_6'
